packages/react-mutation-mapper/src/util/CivicUtils.ts

```diff
@@ -0,0 +1,203 @@
+import * as _ from 'lodash';
+import CivicDataFetcher from "../store/CivicDataFetcher";
+import {ICivicEntry, ICivicGene, ICivicGeneData, ICivicVariant, ICivicVariantData} from "../model/Civic";
+import {Mutation} from "../model/Mutation";
+
+export type MutationSpec = {gene:{hugoGeneSymbol: string}, proteinChange: string};
+
+const civicClient = new CivicDataFetcher();
+
+/**
+ * Asynchronously adds the given variant from a gene to the variant map specified.
+ */
+function addCivicVariant(variantMap: ICivicVariant, variantId: number, variantName: string, geneSymbol: string, geneId: number): Promise<void> {
+    return civicClient.getVariant(variantId, variantName, geneId)
+    .then(function(result: ICivicVariantData) {
+        if (result) {
+            if (!variantMap[geneSymbol]) {
+                variantMap[geneSymbol] = {};
+            }
+            variantMap[geneSymbol][variantName] = result;
+        }
+    });
+}
+
+/**
+ * Asynchronously return a map with Civic information from the genes given.
+ */
+export function getCivicGenes(entrezGeneIds: Array<number>): Promise<ICivicGene> {
+    
+    let civicGenes: ICivicGene = {};
+
+    // Assemble a list of promises, each of which will retrieve a batch of genes
+    let promises: Array<Promise<Array<ICivicGeneData>>> = [];
+    let ids: Array<number> = [];
+    entrezGeneIds.forEach(function(entrezGeneId) {
```

Comment:
this logic is a little confusing.  i would just start by splitting the ids into batches of 400 and then use that list to produce the promises.   then you don't have to invoke getCivicGenesBatch twice

Reply:
agreed. _.chunk would work well for this